File: frontend/src/microservices/attraction/objects/AttractionSuggestionPlannerResponse.ts
// 本文件定义 attraction 模块的 `AttractionSuggestionPlannerResponse`，作为单条建议响应并提供 JSON 编解码。

export type AttractionSuggestionPlannerResponse = {
  resourceType: string
  value: string
  title: string
  subtitle: string
}

export const attractionSuggestionPlannerResponseFromJson = (json: string): AttractionSuggestionPlannerResponse =>
  JSON.parse(json) as AttractionSuggestionPlannerResponse

export const attractionSuggestionPlannerResponseToJson = (value: AttractionSuggestionPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/attraction/objects/AttractionSuggestionPlannerResponse.scala
package com.typesafe.travel.attraction.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class AttractionSuggestionPlannerResponse(resourceType: String, value: String, title: String, subtitle: String)
object AttractionSuggestionPlannerResponse:
  given sourceEncoder: Encoder[AttractionSuggestionPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[AttractionSuggestionPlannerResponse] = deriveDecoder
